AxonDeepSeg/visualization/visualize.py

```diff
@@ -1,174 +1,218 @@
 from tabulate import tabulate
-import numpy as np
+
+# Graphs and plots imports
+from matplotlib.backends.backend_agg import FigureCanvasAgg as FigureCanvas
+from matplotlib.figure import Figure
+
+# AxonDeepSeg imports
 import AxonDeepSeg.ads_utils
+from ..testing.segmentation_scoring import score_analysis, dice
 
 
-def visualize_training(path_model, start_visu=0):
+def visualize_training(path_model, start_for_viz=0):
     """
     :param path_model: path of the folder with the model parameters .ckpt
-    :param start_visu: first iterations can reach extreme values, start_visu set another start than epoch 0
-    :return: evolution
-
-    figure(1) represent the evolution of the loss and the accuracy evaluated on the validation set along the learning process.
-    If the learning began from an initial model, the figure plots first the accuracy and loss evolution from this initial model and then stacks the evolution of the model.
+    :param start_visu: first iterations can reach extreme values, 
+        start_visuset another start than epoch 0
+    :return: (matplotlib.figure.Figure, evolution)
+
+    figure(1) represent the evolution of the loss and the accuracy evaluated on
+    the validation set along the learning process.
+    If the learning began from an initial model, the figure plots first the 
+    accuracy and loss evolution from this initial model and then stacks the 
+    evolution of the model.
     """
 
-    evolution = retrieve_training_data(path_model)
+    def _create_figure_helper(data_evolution):
+        fig = Figure()
+        FigureCanvas(fig)
+
+        # Drawing the evolution curves
+        ax1 = fig.subplots()
+        ax2 = ax1.twinx()
+        ax1.plot(
+            data_evolution["steps"][start_for_viz:],
+            data_evolution["accuracy"][start_for_viz:],
+            "-",
+            label="accuracy",
+        )
+        ax1.set_ylim(ymin=0)
+        ax2.plot(
+            data_evolution["steps"][start_for_viz:],
+            data_evolution["loss"][start_for_viz:],
+            "-r",
+            label="loss",
+        )
+
+        # Annotating the graph
+        ax1.set_title("Accuracy and loss evolution")
+        ax1.set_xlabel("Epoch")
+        ax1.set_ylabel("Accuracy")
+        ax2.set_ylabel("Loss")
+        return fig
 
-    fig = plt.figure(1)
-    # Drawing the evolution curves
-
-    ax = fig.add_subplot(111)
-    ax.plot(evolution['steps'][start_visu:], evolution['accuracy'][start_visu:], '-', label='accuracy')
-    plt.ylabel('Accuracy')
-    plt.ylim(ymin=0)
-    # plt.ylim(ymax=100)
-
-    ax2 = ax.twinx()
-    ax2.plot(evolution['steps'][start_visu:], evolution['loss'][start_visu:], '-r', label='loss')
-
-    # Annotating the graph
-
-    plt.title('Accuracy and loss evolution')
-    plt.ylabel('Loss')
-    plt.xlabel('Epoch')
-    plt.show()
+    evolution = retrieve_training_data(path_model)
+    fig = _create_figure_helper(evolution)
 
-    return evolution
+    return fig, evolution
```

Comment:
Maybe we should keep `evolution` as the first returned value and have `fig`, in case someone's code relies on this one (and that keeps the fig optional). What do you think?

Reply:
I modified files accordingly. Actually, it's not very good practices to have a function called `visualize` (or `draw`, or etc.) that returns data. It should simply draws or in our case return a figure object.